snmp/datadog_checks/snmp/models.py

```diff
@@ -0,0 +1,144 @@
+    """
+
+    def __init__(
+        self, obj  # type: typing.Union[str, typing.Tuple[int, ...], ObjectIdentity, ObjectName]
+    ):
+        # type: (...) -> None
+        identifier = ()  # type: typing.Tuple[int, ...]
+
+        if isinstance(obj, str):
+            identifier = tuple(map(int, obj.lstrip('.').split('.')))
+        elif isinstance(obj, tuple):
+            identifier = obj
+        elif isinstance(obj, ObjectIdentity):
+            try:
+                identifier = obj.getOid().asTuple()
+            except SmiError as exc:
+                raise RuntimeError(
+                    'Could not retrieve OID from `ObjectIdentity`: {!r}.\n'
+                    'HINT: did you try passing an `ObjectIdentity` that was built manually? '
+                    'This will only work it has been resolved, i.e. if you or someone else called '
+                    '`.resolveWithMib(mibViewController)` on it. Note that this should have been done'
+                    'done by PySNMP if this `ObjectIdentity` was obtained by executing an SNMP command.'.format(exc)
+                )
+        elif isinstance(obj, ObjectName):
+            identifier = obj.asTuple()
+        else:
+            raise TypeError('Building an OID from object {!r} of type {} is not supported'.format(obj, type(obj)))
+
+        self._identifier = identifier
+
+    def as_tuple(self):
+        # type: () -> typing.Tuple[int, ...]
+        return self._identifier
+
+    def as_string(self):
+        # type: () -> str
+        return '.'.join(map(str, self.as_tuple()))
+
+    def as_object_identity(self):
+        # type: () -> ObjectIdentity
+        return ObjectIdentity(self.as_tuple())
+
+    def as_object_type(self):
+        # type: () -> ObjectType
+        return ObjectType(self.as_object_identity())
+
+    def __repr__(self):
+        # type: () -> str
+        return 'OID({!r})'.format(self.as_string())
+
+
+class Variable(object):
+    """
+    Represents the response of the SNMP host to a given requested OID.
+
+    Also known as a 'MIB variable', or a 'var_bind' in the PySNMP jargon.
+    """
+
+    def __init__(self, var_bind):
+        # type: (typing.Union[ObjectType, typing.Tuple]) -> None
+        try:
+            self.name, self.value = var_bind
+        except SmiError as exc:
+            raise RuntimeError(
+                'Could not destructure variable: {!r}.\n'
+                'HINT: Did you try instanciating a `Variable()` object directly? '
+                'This can only work if the given `ObjectType` has been resolved, i.e. someone '
+                'called `.resolveWithMib(mibViewController)` on it. Note that this should have been done by '
+                'PySNMP if this variable came from executing an SNMP command.'.format(exc)
+            )
+        else:
+            self.var_bind = var_bind
+
+    @property
+    def oid(self):
+        # type: () -> OID
+        return OID(self.name)
```

Comment:
Would it make sense to make the assignment at init time in case there are errors initializing the OID?

Reply:
Good idea, yup. Actually I replaced `self.name` with `self.oid`, because that's mostly what we care about here.